torch/utils/_config_module.py

```diff
@@ -407,14 +407,36 @@ def _set_alias_val(self, entry: _ConfigEntry, val: Any) -> None:
         module, constant_name = data
         setattr(module, constant_name, val)
 
-    def _is_default(self, name: str) -> bool:
+    def _is_default(self, name: str, env: bool = False) -> bool:
+        """
+        If env=True, configs overriden by the env are not considered default.
+        """
+        if env:
+            config_val = self._config[name]
+            # The config is not overridden by the user, and the env_value_default
+            # is different from the default value (meaning user has set the env to
+            # change the default value).
+            not_set_env_default = (
+                config_val.env_value_default is _UNSET_SENTINEL
+                or config_val.env_value_default == config_val.default
+            )
+            not_set_env_force = (
+                config_val.env_value_force is _UNSET_SENTINEL
+                or config_val.env_value_force == config_val.default
+            )
+            return (
+                config_val.user_override is _UNSET_SENTINEL
+                and not_set_env_default
+                and not_set_env_force
+            )
         return self._config[name].user_override is _UNSET_SENTINEL
 
     def _get_dict(
         self,
         ignored_keys: Optional[List[str]] = None,
         ignored_prefixes: Optional[List[str]] = None,
         skip_default: bool = False,
+        env_default: bool = False,
```

Comment:
Does anything break if we just turn this on always?  If so let's just remove the option and make it default.

Reply:
@c00w Do you know if we can make this `env_default` default to True? (If a config's env value is different from default value, it cannot be skipped by `skip_default`).

Currently, the test is written such that these configs are skipped by `skip_default`.

Reply:
I think it's probably ok. I'll make it default then.